crates/node/src/command.rs

```diff
@@ -83,41 +66,24 @@ fn set_dev_environment(cli: &mut Cli) {
     cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
 }
 
-fn set_chain_spec(cli: &mut Cli) -> Result<(), String> {
+fn set_testnet(cli: &mut Cli, testnet: Testnet) -> Result<(), String> {
     // checks if it should retrieve and enable a specific chain-spec
     let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
     let local_path = utils::get_project_path();
 
-    if let Some(chain_spec_url) = cli.run.chain_spec_url.clone() {
-        utils::fetch_from_url(chain_spec_url.clone(), madara_path.clone() + "/chain-specs", cli.run.update_configs)?;
-        let chain_spec = chain_spec_url.split('/').last().expect("Chain spec file name not found");
-        cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/" + chain_spec);
-
-        return Ok(());
-    }
-
-    match (cli.run.testnet, cli.run.disable_url_fetch, cli.run.disable_madara_configs, local_path) {
-        (Some(Testnet::Sharingan), _, false, Ok(ref src_path)) => {
+    match (testnet, local_path) {
+        (Testnet::Sharingan, Ok(ref src_path)) => {
             let src_path = src_path.clone() + "/configs/chain-specs/testnet-sharingan-raw.json";
-            utils::copy_from_filesystem(src_path, madara_path.clone() + "/chain-specs", cli.run.update_configs)?;
+            utils::copy_from_filesystem(src_path, madara_path.clone() + "/chain-specs")?;
             cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
-        (Some(Testnet::Sharingan), false, false, _) => {
-            let madara_configs: configs::Configs =
-                serde_json::from_str(&utils::read_file_to_string(madara_path.clone() + "/configs/index.json")?)
-                    .expect("Failed to load madara configs");
-            // If testnet flag is specified, right now it's only Sharingan
-            for chain_spec in madara_configs.chain_specs {
-                configs::fetch_and_validate_file(
-                    madara_configs.remote_base_path.clone(),
-                    chain_spec,
-                    madara_path.clone() + "/configs/chain-specs/",
-                    cli.run.update_configs,
-                )?;
-            }
+        (Testnet::Sharingan, _) => {
+            utils::fetch_from_url(
+                constants::SHARINGAN_CHAIN_SPEC_URL.to_string(),
+                madara_path.clone() + "/configs/chain-specs/",
+            )?;
             cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
-        _ => {}
     }
 
     if cli.run.run_cmd.shared_params.chain.is_some() {
```

Comment:
Is this compatible with `Testnet` can you have both `--testnet` and `--chain`?

Reply:
you right